### PR TITLE
Fixing absTime bugs

### LINKED ISSUE #527: If absTime is deleted, it sets absTime as zero and disables it instead
Screen Shot 2021-08-03 at 1 24 56 PM

### LINKED ISSUE #530: If there is an absolute length, the repeating starting segment must have an absTime
image

### PR DESCRIPTION
**Pull request by i-m-foster**
Closes #527 and #530

### CODE REVIEW

File: leaf-ui/rappid-extensions/ElementInspector.js [hunk 1]
```diff
@@ -418,9 +420,11 @@ var ElementInspector = Backbone.View.extend({
      * (the select elements for repeat begin and end)
      */
     selectRepeatValues: function () {
-        // If some of the repeat select values were previously disbaled, enable them
+        // If some of the repeat select values were previously disabled, enable them
         this.$("option").prop('disabled', '');
```
Comment (Domily11) [inline leaf-ui/rappid-extensions/ElementInspector.js:430]: Maybe delete this line? Up to you tho

File: leaf-ui/rappid-extensions/ElementInspector.js [hunk 2]
```diff
@@ -471,10 +489,28 @@ var ElementInspector = Backbone.View.extend({
...
         if (absLength < 0) {
             $('#repeat-end3').val(0);
         }
         this.intention.get('evolvingFunction').set('repAbsTime', absLength);
+
+        // For when user selects starting repeat value prior to adding absolute length
+        // find the startAT value for the starting repeating function segment
+        for (var i = 0; i<this.intention.getFuncSegments().length; i++) {
+            if (this.intention.getFuncSegments()[i].get('startTP') == begin) {
+                console.log('hi');
```
Comment (Domily11) [inline leaf-ui/rappid-extensions/ElementInspector.js]: Remove console

**Review by Domily11 — CHANGES_REQUESTED** [review]
Small additional bug, the error doesn't pop back up if you added the absolute time then immediately took it off

File: leaf-ui/rappid-extensions/ElementInspector.js [hunk 3]
```diff
@@ -439,9 +444,30 @@ var ElementInspector = Backbone.View.extend({
...
+
+        // Once a repeating start value is selected, rerender function segments to disable absTime
+        if ($("#repeat-begin").val() != null) {
+            this.rerender();
+        }
+
         if (begin === null || end === null) {
             return;
-        }
+        }
```
Comment (Domily11) [inline leaf-ui/rappid-extensions/ElementInspector.js]: Delete extra space

**Review by YesugenB — APPROVED** [review]

**Review by Domily11 — CHANGES_REQUESTED** [review]
One bug:
- if you chose a begin without putting an absolute time and the error message appears, the message doesn't disappear even tho the begin is emptied

**Review by Domily11 — APPROVED** [review]

**Comment by amgrubb** [comment]
It still not working quite correctly. Maybe we can push this as a partial fix and revisit in the future.

**Review by amgrubb — APPROVED** [review]
It's still not working quite correctly. Maybe we can push this as a partial fix and revisit in the future.
If so, add a new issue to improve the interaction of the error message.

**Pull request merged**